OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java

```diff
@@ -568,7 +570,14 @@ public boolean search(final SearchPhrase phrase, final SearchResultMatcher resul
 					SearchPhraseDataType.POI);
 			String searchWord = phrase.getUnknownWordToSearch();
```

Comment:
searchWord for OLC here

Reply:
It can’t be using here. It contains cityname. Not OLC code.
```

searchWord = “Kyiv”
LocationParser.isValidOLC(searchWord) = false
phrase.getFirstUnknownSearchWord() = “CGP5+95”
LocationParser.isValidOLC(phrase.getFirstUnknownSearchWord() ) = true
```